### PR TITLE
Implement permissions

### PR DESCRIPTION
**Pull request by ShrawanBK**

### CODE REVIEW

File: src/components/CountrySummary/index.tsx [hunk 1]
```diff
@@ -37,11 +38,17 @@ function CountrySummary(props: CountrySummaryProps) {
...
+        add: user?.permissions?.add?.summary,
+        modify: user?.permissions?.change?.summary,
+    };
+
     return (
         <Container
             className={_cs(className, styles.summary)}
             heading="Summary"
-            headerActions={(
+            headerActions={(summaryPermission.add || summaryPermission.modify) && (
```
Comment (tnagorra) [inline src/components/CountrySummary/index.tsx:51]: I don't think we should have `summaryPermission.modify` here. @annshress haina ra?

File: src/components/ContextualUpdate/index.tsx [hunk 2]
```diff
@@ -50,13 +58,15 @@ function ContextualUpdate(props: CountryContextualUpdateProps) {
...
                         <IoMdDownload />
                     </QuickActionButton>
-                    <QuickActionButton
-                        name={undefined}
-                        disabled={disabled}
-                        onClick={handleContextualFormOpen}
-                    >
-                        {contextualUpdate ? <IoMdCreate /> : <IoMdAdd />}
-                    </QuickActionButton>
+                    {(contextualPermission.add || contextualPermission.modify) && (
```
Comment (tnagorra) [inline src/components/ContextualUpdate/index.tsx:61]: I think we should not have `contextualPermission.modify` here. @annshress haina ra?

File: src/components/CountrySummary/index.tsx (hunk 1 above)
Comment (annshress) [reply src/components/CountrySummary/index.tsx:51]: Yes.

File: src/components/ContextualUpdate/index.tsx (hunk 2 above)
Comment (annshress) [reply src/components/ContextualUpdate/index.tsx:61]: Yes. These are always added.

Comment (ShrawanBK) [reply src/components/ContextualUpdate/index.tsx:61]: okay !!

**Review by tnagorra — CHANGES_REQUESTED** [review]
We will need to transform the permissions data structure so that we don't have to transform it everytime we use it. The transformation can either be done on the client side or the server side.

**Pull request merged**